replications/internal/queue_management.go

```diff
@@ -114,44 +119,44 @@ func (rq *replicationQueue) Close() error {
 	return rq.queue.Close()
 }
 
+// WriteFunc is currently a placeholder for the "default" behavior
+// of the queue scanner sending data from the durable queue to a remote host.
+func WriteFunc(b []byte) error {
+	return nil
+}
+
 func (rq *replicationQueue) run() {
 	defer rq.wg.Done()
 
-	retryInterval := time.Second
-	retryTimer := time.NewTicker(retryInterval)
-	defer retryTimer.Stop()
-
-	writer := func() {
-		for {
-			_, err := rq.SendWrite(func(b []byte) error {
-				rq.logger.Info("written bytes", zap.Int("len", len(b)))
-				return nil
-			})
-			if err != nil {
-				if err == io.EOF {
-					// No more data
-					// Handle this gracefully, as it is an expected error to receive
-				} else {
-					// todo more error handling
-					panic(1)
-				}
-				break
-			}
-		}
-	}
-
 	for {
 		select {
 		case <-rq.done: // end the goroutine when done is messaged
 			return
-		case <-retryTimer.C: // run the scanner every 1s
-			writer()
 		case <-rq.receive: // run the scanner on data append
-			writer()
+			for {
+				_, err := rq.SendWrite(rq.writeFunc)
+
+				if err != nil {
+					if err == io.EOF {
+						// No more data
+						// Handle this gracefully, as it is an expected error to receive
+					} else {
+						// Crash out the server if an unhandleable error occurs in SendWrite()
+						// Should ideally never be hit
+						panic(1)
```

Comment:
A better approach IMO would be to push all error-handling into `SendWrite`, and adjust its signature to match. That way we don't have to figure out a last-ditch thing to do with errors here.

My suggestion would be:
```
func (rq *replicationQueue) SendWrite(dp func([]byte) error) bool
```
Where a return value of `false` is the signal for the outer loop to terminate. This would also push the responsibility of checking the EOF error into `SendWrite` - I don't think this is too bad, since the code there is already doing some checks on EOF.